### DISCUSSION TITLE
Unable to login after deploy

### DISCUSSION THREAD

**Discussion by lianlun1** [prisma/prisma#13760] [Q&A] [answered]
I make the application exactly according to the instructions https://vercel.com/guides/nextjs-prisma-postgres, but after deploy, authorization does not work. An error appears on the login page: 

> Server error
There is a problem with the server configuration.
Check the server logs for more information.

An error appears in the logs:

> 2022-06-11T13:42:03.145Z	0bf74056-38ef-48b2-b9c2-b971c23fb1a6	ERROR	[next-auth][error][NO_SECRET] 
https://next-auth.js.org/errors#no_secret Please define a `secret` in production. MissingSecret [MissingSecretError]: Please define a `secret` in production.
    at assertConfig (/var/task/node_modules/next-auth/core/lib/assert.js:26:14)
    at NextAuthHandler (/var/task/node_modules/next-auth/core/index.js:34:52)
    at NextAuthNextHandler (/var/task/node_modules/next-auth/next/index.js:21:51)
    at NextAuth (/var/task/node_modules/next-auth/next/index.js:60:10)
    at authHandler (/var/task/.next/server/pages/api/auth/[...nextauth].js:56:61)
    at Object.apiResolver (/var/task/node_modules/next/dist/server/api-utils.js:101:15)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async NextNodeServer.runApi (/var/task/node_modules/next/dist/server/next-server.js:320:9)
    at async Object.fn (/var/task/node_modules/next/dist/server/base-server.js:486:37)
    at async Router.execute (/var/task/node_modules/next/dist/server/router.js:228:32) {
  code: 'NO_SECRET'
}

Authorization works on localhost.

**Comment by janpio** [comment, marked as answer]
Did you provide the env vars as secrets as defined in https://vercel.com/guides/nextjs-prisma-postgres#step-9.-deploy-to-vercel? (More direct link https://vercel.com/guides/nextjs-prisma-postgres#step-9.-deploy-to-vercel:~:text=Here%27s%20what%20you%20need%20to%20provide%3A)

**Reply by lianlun1** [reply]
Right. All data running on localhost is transferred accurately.

**Reply by janpio** [reply]
Ah, I think this is actually a Next-Auth error message as linked in the error message: https://next-auth.js.org/errors#no_secret

> **NO_SECRET**
> In production, we expect you to define a secret property in your configuration. In development, this is shown as a warning for convenience. [Read more](https://next-auth.js.org/configuration/options#secret)

Maybe that is missing in the tutorial explicitly?

**Reply by janpio** [reply]
I see here that the code should take it from the ENV: https://vercel.com/guides/nextjs-prisma-postgres#step-9.-deploy-to-vercel:~:text=secret%3A%20process.env.SECRET%2C (`  secret: process.env.SECRET,`) So maybe try to set `SECRET` of any value via the env var UI on Vercel and see if this fixes it.

**Reply by lianlun1** [reply]
If you mean it https://i.stack.imgur.com/hRJZK.png, I have already registered all the data here.

**Reply by lianlun1** [reply]
Since the .env file is added to .gitignore (which is correct), I added these values to settings/environment-variables in the Vercale interface

**Reply by janpio** [reply]
You are lacking an environment variable with the (bad) name `SECRET`. That is what the error message is complaining about. Please try to add it (I think the value does not matter much, so just put a random string in there) and see if this fixes the problem.

**Reply by lianlun1** [reply]
No, it doesn't help.

**Reply by janpio** [reply]
Can you share the current state of your repository? Happy to take a look if I can deploy.

**Reply by lianlun1** [reply]
https://github.com/lianlun1/alex-antonov-blogr-nextjs-prisma

**Reply by janpio** [reply]
After naive deployment of the code, I get a server error as you describe: https://alex-antonov-blogr-nextjs-prisma-ew8e6m42b-janpio.vercel.app/api/auth/signin
Then I added an env var `SECRET` with value `foo` for all environments, redeployed, and it works: https://alex-antonov-blogr-nextjs-prisma-agt62noy0-janpio.vercel.app/api/auth/signin

**Reply by lianlun1** [reply]
Now I have an authorization button, but when I click on it, a message appears: Unable to sign in. Github does not redirect to the authorization page itself.

**Reply by lianlun1** [reply]
It seems that these are already problems with github, and not with my project.

**Reply by janpio** [reply]
That would then possibly be a problem with the env vars for GitHub auth, `GITHUB_ID` and `GITHUB_SECRET` - I did not try to set that up for my project at all. Let me try.

**Reply by janpio** [reply]
Yep, seems to be working fine when I deploy to production and use that URL to open the page: https://alex-antonov-blogr-nextjs-prisma-lemon.vercel.app/

**Reply by lianlun1** [reply]
Yes, your solution is correct, I didn't notice that the url changed after redeploy. Now that I've fixed it, everything works. Although I still do not understand why authorization was opened with my friend, while changing my ip did not help.

**Reply by dorin-js** [reply]
> You are lacking an environment variable with the (bad) name `SECRET`. That is what the error message is complaining about. Please try to add it (I think the value does not matter much, so just put a random string in there) and see if this fixes the problem.

Hey, janpio, many thanks for the help. I just faced same problem and solved it thanks to you.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
